core/src/test/java/org/apache/brooklyn/entity/stock/BasicStartableTest.java

```diff
@@ -148,11 +149,14 @@ public void testIgnoresUnstartableEntities() throws Exception {
     @Test
     public void testTransitionsThroughLifecycles() throws Exception {
         startable = app.addChild(EntitySpec.create(BasicStartable.class));
+        EntityAsserts.assertAttributeEqualsEventually(app, Attributes.SERVICE_STATE_ACTUAL, Lifecycle.STOPPED);
+        
         RecordingSensorEventListener<Lifecycle> listener = new RecordingSensorEventListener<Lifecycle>(true);
         managementContext.getSubscriptionContext(startable)
                 .subscribe(startable, Attributes.SERVICE_STATE_ACTUAL, listener);
 
         app.start(ImmutableList.of(loc1));
+        app.config().set(StartableApplication.DESTROY_ON_STOP, false);
```

Comment:
The assertion below this one looks like it would fail non-deterministically. 

We have added a listener (with `managementContext.getSubscriptionContext(startable).subscribe(...)`, so that will get notifications asynchronously. Immediately after calling `app.stop()`, we assert that the listener has the expected events. I'd have though we'd need to use `Asserts.succeedsEventually(...)` when checking if the listener has received the expected notifications.